flytepropeller/pkg/compiler/transformers/k8s/workflow.go

```diff
@@ -231,7 +252,25 @@ func BuildFlyteWorkflow(wfClosure *core.CompiledWorkflowClosure, inputs *core.Li
 		errs.Collect(errors.NewWorkflowBuildError(err))
 	}
 
-	obj.ObjectMeta.Name = name
+	if workflowCRNameHashLength := config.GetConfig().WorkflowCRNameHashLength; workflowCRNameHashLength > 0 {
+		// Seed the randomness before generating the name with random suffix
+		hashedIdentifier := hashIdentifier(core.Identifier{
+			Project: project,
+			Domain:  domain,
+			Name:    name,
+		})
+		rand.Seed(int64(hashedIdentifier))
+
+		base := name + "-"
+		maxNameLength := allowedExecutionNameLength - workflowCRNameHashLength
+		if len(base) > maxNameLength {
+			base = base[:maxNameLength]
+		}
+		obj.ObjectMeta.Name = fmt.Sprintf("%s%s", base, rand.String(workflowCRNameHashLength))
```

Comment:
Based off of https://github.com/clarklee92/kubernetes/blob/519f1ba07379108c301c0474d5eb61b2012bd427/staging/src/k8s.io/apiserver/pkg/storage/names/generate.go#L57-L63

Reply:
A fixed suffix length of 10 sounds good to me. Agreed about the sufficient amount of entropy on relatively high values for the suffix length. https://github.com/flyteorg/flyte/pull/5480/commits/15b57cb739f74cc00e133db79f19301b87d5f726